os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -560,7 +560,12 @@ void binary_manager_release_binary_sem(int bin_idx)
 				if (holder && holder->htcb && holder->htcb->group && holder->htcb->group->tg_binidx == bin_idx) {
 					/* Increase semcount and release itself from holder */
 					sem->semcount++;
-					sem_freeholder(sem, holder);
+#ifdef CONFIG_PRIORITY_INHERITANCE
+					if ((sem->flags & PRIOINHERIT_FLAGS_DISABLE) != 0)
```

Comment:
I think we should move this code into the sem_unblock_task.
Because, in normal case, there is no free in sem_post. In this case, holder will be remained.
And you'd better remove the destroyholder in sem_setprotocol, because we should trace that sem also.